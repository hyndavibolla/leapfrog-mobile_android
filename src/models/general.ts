import { Dispatch } from 'react';
import Auth0 from 'react-native-auth0';

import { ApiService } from '_services/ApiService';
import { AppService } from '_services/AppService';
import { CCPAService } from '_services/CCPAService';
import { ICrashlyticsService } from '_services/CrashlyticsService';
import { EventTrackerService } from '_services/EventTrackerService';
import { IHttpService } from '_services/HttpService';
import { NativeHelperService } from '_services/NativeHelperService';
import { Logger } from '_services/Logger';
import { IPerformanceTrackingService } from '_services/PerformanceTrackingService';
import { RaiseService } from '_services/RaiseService';
import { RemoteConfigService } from '_services/RemoteConfigService';
import { ICardLinkState } from '_state_mgmt/cardLink/state';
import { ICoreState } from '_state_mgmt/core/state';
import { IGameState } from '_state_mgmt/game/state';
import { IGiftCardState } from '_state_mgmt/giftCard/state';
import { IMessagesState } from '_state_mgmt/messages/state';
import { IMissionState } from '_state_mgmt/mission/state';
import { IRewardState } from '_state_mgmt/reward/state';
import { ISurveyState } from '_state_mgmt/survey/state';
import { IUserState } from '_state_mgmt/user/state';

export interface IAction<T = string, P = any> {
  type: T;
  payload: P;
}

export type IReducer<S = any, A = any> = (state: S, action: IAction<A>) => S;

export interface Deps {
  apiService: ApiService;
  appService: AppService;
  crashlyticsService: ICrashlyticsService;
  httpService: IHttpService;
  performanceTrackingService: IPerformanceTrackingService;
  raiseService: RaiseService;
  nativeHelperService: NativeHelperService;
  stateSnapshot: { get: () => IGlobalState; set: (state: IGlobalState) => void };
  logger: Logger;
  eventTrackerService: EventTrackerService;
  ccpaService: CCPAService;
  remoteConfigService: RemoteConfigService;
  authService: Auth0;
}
export type IGlobalState = {
  core: ICoreState;
  user: IUserState;
  game: IGameState;
  mission: IMissionState;
  reward: IRewardState;
  survey: ISurveyState;
  messages: IMessagesState;
  cardLink: ICardLinkState;
  giftCard: IGiftCardState;
};

export interface IGlobalContext {
  state: IGlobalState;
  dispatch: Dispatch<IAction<any, any>>;
  deps: Deps;
}

export enum HTTP_METHOD {
  GET = 'GET',
  POST = 'POST',
  PUT = 'PUT',
  PATCH = 'PATCH',
  DELETE = 'DELETE'
}

export interface ITokenSet {
  accessToken: string;
  refreshToken: string;
  accessTokenExpiryTime: string;
  sywToken?: string;
  /**
   * @deprecated deviceId does not belong to ITokenSet anymore.
   */
  deviceId?: string;
  scope?: string;
  expiresIn?: number;
  tokenType?: string;
}

export enum FeatureFlag {
  API_OVERRIDE = 'API_OVERRIDE',
  TEST_ONLY = 'TEST_ONLY',
  ENV_BANNER = 'ENV_BANNER',
  BUTTON_DEBUG = 'BUTTON_DEBUG',
  LOG_LEVEL_INFO = 'LOG_LEVEL_INFO',
  LOG_LEVEL_DEBUG = 'LOG_LEVEL_DEBUG',
  LOG_LEVEL_WARN = 'LOG_LEVEL_WARN',
  LOG_LEVEL_ERROR = 'LOG_LEVEL_ERROR',
  LOG_VIEW = 'LOG_VIEW',
  STREAK_REDEEM_CARD = 'STREAK_REDEEM_CARD',
  SAILTHRU_TRACKING = 'SAILTHRU_TRACKING',
  SURVEY = 'SURVEY',
  PREDICTIVE_MAP_SEARCH = 'PREDICTIVE_MAP_SEARCH',
  MISSIONS_SEARCH = 'MISSIONS_SEARCH',
  SEE_ALL_TRANSACTIONS = 'SEE_ALL_TRANSACTIONS',
  WALLET_YOUR_GC = 'WALLET_YOUR_GC'
}

export type DateLike = Date | string | number;

export type HTMLString = string;

export interface IGameLevelUI {
  updatedAt: DateLike;
  level: number;
}

export interface ICCPASetting {
  allow: boolean;
}

export interface IOnboarding {
  seenOnboarding: boolean;
}

export enum ButtonCreativeType {
  HERO = 'hero',
  CAROUSEL = 'carousel',
  LIST = 'list',
  GRID = 'grid',
  DETAIL = 'detail',
  OTHER = 'other'
}

export interface ISearchHistoryItem {
  keyword: string;
  lastUpdatedAt: DateLike;
}

export enum TooltipKey {
  EARN = 'earn',
  REWARDS = 'rewards',
  MISSIONS = 'missions',
  STREAK = 'streak'
}

export interface IApiOverrideRule {
  key: string;
  description?: string;
  matcher: string; // reg exp
  status: number;
  response: any;
}

export namespace RemoteConfigModel {
  export interface IAppVersion {
    suggested: number;
    required: number;
  }

  export interface IEarnSection {
    name: string;
    order: number;
    visible: boolean;
    title: string;
    description: string;
    seeAllButton: boolean;
  }
}

export enum RUNTIME_ENV {
  LIVE = 'live',
  DEV_TEST = 'test'
}

export interface IMeasure {
  x: number;
  y: number;
  width: number;
  height: number;
}

export interface IExpirePointsBannerDataSet {
  lastPointsDate: DateLike;
  lastAcceptedDate: DateLike;
}

export interface ISailthruMessage {
  createdAt: string;
  custom: {
    [k: string]: string;
  };
  htmlText: string;
  id: string;
  isRead: boolean;
  text: string;
  title: string;
  type: string;
  cardImageUrl?: string;
  share: boolean;
}

export interface ITutorialStatus {
  isBannerWatched: boolean;
  date: string;
}
