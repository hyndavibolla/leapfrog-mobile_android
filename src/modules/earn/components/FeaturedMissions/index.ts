export { default as FeaturedMissions } from './FeaturedMissions';
