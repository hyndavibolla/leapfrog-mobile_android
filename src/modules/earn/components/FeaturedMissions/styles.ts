import { css } from '@emotion/native';

export const styles = {
  sectionMain: css`
    padding-horizontal: 16px;
    padding-bottom: 24px;
  `,
  streakSectionHeader: css`
    padding-bottom: 16px;
    justify-content: space-between;
  `
};
