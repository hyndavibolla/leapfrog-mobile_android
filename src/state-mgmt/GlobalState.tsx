import React, { createContext, memo, PropsWithChildren, useReducer, Dispatch, useEffect } from 'react';

import { IGlobalState, Deps, IReducer, IGlobalContext } from '_models/general';
import { reducer as coreReducer } from './core/reducer';
import { reducer as userReducer } from './user/reducer';
import { reducer as gameReducer } from './game/reducer';
import { reducer as missionReducer } from './mission/reducer';
import { reducer as rewardReducer } from './reward/reducer';
import { reducer as surveyReducer } from './survey/reducer';
import { reducer as messagesReducer } from './messages/reducer';
import { reducer as cardLinkReducer } from './cardLink/reducer';
import { reducer as giftCardReducer } from './giftCard/reducer';
import { initialState as coreInitialState } from './core/state';
import { initialState as userInitialState } from './user/state';
import { initialState as gameInitialState } from './game/state';
import { initialState as missionInitialState } from './mission/state';
import { initialState as rewardInitialState } from './reward/state';
import { initialState as surveyInitialState } from './survey/state';
import { initialState as messagesInitialState } from './messages/state';
import { initialState as cardLinkInitialState } from './cardLink/state';
import { initialState as giftCardInitialState } from './giftCard/state';
import { combineReducers } from '_utils/combineReducers';
import { GeneralModel } from '_models';
import { deepClone } from '_utils/deepClone';

export const getInitialState = () =>
  deepClone<IGlobalState>({
    core: coreInitialState,
    user: userInitialState,
    game: gameInitialState,
    mission: missionInitialState,
    reward: rewardInitialState,
    survey: surveyInitialState,
    messages: messagesInitialState,
    cardLink: cardLinkInitialState,
    giftCard: giftCardInitialState
  });

export const combinedReducer = combineReducers<GeneralModel.IGlobalState>({
  core: coreReducer,
  user: userReducer,
  game: gameReducer,
  mission: missionReducer,
  reward: rewardReducer,
  survey: surveyReducer,
  messages: messagesReducer,
  cardLink: cardLinkReducer,
  giftCard: giftCardReducer
});

export const GlobalContext = createContext<{ state: IGlobalState; dispatch: Dispatch<any>; deps: Deps }>({
  state: getInitialState(),
  dispatch: /* istanbul ignore next */ () => null,
  deps: null
} as IGlobalContext);

export const GlobalProvider = memo(
  ({
    children,
    deps,
    initState,
    combinedReducers
  }: PropsWithChildren<{
    deps: Deps;
    initState: IGlobalState;
    combinedReducers: IReducer<IGlobalState>;
  }>) => {
    const [state, dispatch] = useReducer(combinedReducers, initState);
    useEffect(() => deps.stateSnapshot.set(state), [state]); // eslint-disable-line react-hooks/exhaustive-deps
    return <GlobalContext.Provider value={{ state, dispatch, deps } as IGlobalContext}>{children}</GlobalContext.Provider>;
  }
);
