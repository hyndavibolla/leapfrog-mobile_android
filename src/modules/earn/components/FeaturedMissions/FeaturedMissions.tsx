import React, { memo, useEffect } from 'react';
import { View } from 'react-native';
import { useNavigation } from '@react-navigation/native';

import { Title, TitleType } from '_components/Title';
import ErrorBoundary from '_components/ErrorBoundary';
import { LargeStreakCard } from '_components/LargeStreakCard/LargeStreakCard';
import { SectionError } from '../SectionError';

import { useTestingHelper } from '_utils/useTestingHelper';
import { useGetStreakList } from '_state_mgmt/streak/hooks';
import { ROUTES } from '_constants';

import { styles } from './styles';

export interface Props {
  focusKey: string;
  isLoadingCallback: (isLoading: boolean) => void;
  title?: string;
  description?: string;
  seeAllButton?: boolean;
}

export const FeaturedMissions = ({ focusKey, isLoadingCallback, ...props }: Props) => {
  const { getTestIdProps } = useTestingHelper('earn-main-featured-missions');
  const navigation = useNavigation();
  const [onGetStreakList, isLoadingStreakList = true, streakListError, streakList = []] = useGetStreakList();

  useEffect(() => {
    onGetStreakList(true);
  }, [focusKey, onGetStreakList]);

  useEffect(() => {
    isLoadingCallback(isLoadingStreakList);
  }, [isLoadingStreakList, isLoadingCallback]);

  return streakListError ? (
    <SectionError />
  ) : !streakList.length ? null : (
    <View style={styles.sectionMain} {...getTestIdProps('container')}>
      <ErrorBoundary>
        <View style={styles.streakSectionHeader} {...getTestIdProps('streak-header')}>
          <Title type={TitleType.SECTION} numberOfLines={1} ellipsizeMode="tail">
            {props?.title ?? 'Featured Missions'}
          </Title>
        </View>
        {streakList.slice(0, 1).map(streak => (
          <View key={streak.rewardOfferCode}>
            <LargeStreakCard streak={streak} onPress={() => navigation.navigate(ROUTES.MAIN_TAB.STREAK)} />
          </View>
        ))}
      </ErrorBoundary>
    </View>
  );
};

export default memo(FeaturedMissions);
