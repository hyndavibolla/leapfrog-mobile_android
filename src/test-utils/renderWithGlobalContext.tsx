import React, { ReactElement } from 'react';
import { render } from '@testing-library/react-native';

import { getInitialState, GlobalProvider } from '../state-mgmt/GlobalState';
import { Deps, IGlobalState, IReducer } from '../models/general';
import { getMockDeps } from './getMockDeps';
import { getMockReducer } from './getMockReducer';

export const renderWithGlobalContext = (
  children: ReactElement,
  deps: Deps = getMockDeps(),
  initState: IGlobalState = getInitialState(),
  combinedReducers: IReducer<IGlobalState, any> = getMockReducer()
) => {
  const renderAPI = render(
    <GlobalProvider deps={deps} initState={initState} combinedReducers={combinedReducers}>
      {children}
    </GlobalProvider>
  );
  return { ...renderAPI, mockReducer: combinedReducers, deps };
};
