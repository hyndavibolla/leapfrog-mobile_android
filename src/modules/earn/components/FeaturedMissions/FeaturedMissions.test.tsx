import React from 'react';
import { fireEvent } from '@testing-library/react-native';
import { act } from 'react-test-renderer';

import { ROUTES } from '_constants';
import { Deps, IGlobalState } from '_models/general';
import { getStreak_1 } from '_test_utils/entities';
import { getInitialState } from '_state_mgmt/GlobalState';
import { getMockDeps } from '_test_utils/getMockDeps';
import { renderWithGlobalContext } from '_test_utils/renderWithGlobalContext';
import { createUUID } from '_utils/create-uuid';
import { wait } from '_utils/wait';
import FeaturedMissions, { Props } from './FeaturedMissions';

const mockNavigate = jest.fn();
jest.mock('@react-navigation/native', () => ({
  __esModule: true,
  useNavigation: () => ({ navigate: mockNavigate })
}));

describe('FeaturedMissions', () => {
  let props: Props;
  let deps: Deps;
  let initialState: IGlobalState;

  beforeEach(() => {
    deps = getMockDeps();
    initialState = getInitialState();
    props = {
      focusKey: createUUID(),
      isLoadingCallback: jest.fn()
    };
  });

  it('should render', async () => {
    deps.apiService.fetchStreakList = jest.fn().mockResolvedValue([{ ...getStreak_1(), currentQualifiedValue: 3, thresholdValue: 3 }]);
    const { toJSON } = renderWithGlobalContext(<FeaturedMissions {...props} />, deps, initialState);
    await act(() => wait(0));
    expect(toJSON()).toMatchSnapshot();
  });

  it('on mission click navigate to missions screen', async () => {
    const { queryAllByTestId } = renderWithGlobalContext(<FeaturedMissions {...props} />, deps, initialState);
    await act(() => wait(0));
    fireEvent.press(queryAllByTestId('large-streak-card-container')[0]);
    expect(mockNavigate).toBeCalledWith(ROUTES.MAIN_TAB.STREAK);
  });

  it('should refresh data by default', async () => {
    deps.apiService.fetchStreakList = jest.fn().mockResolvedValue([{ ...getStreak_1(), currentQualifiedValue: 3, thresholdValue: 3 }]);
    initialState.core.routeHistory = [ROUTES.MAIN_TAB.EARN, ROUTES.MAIN_TAB.WALLET];
    const { toJSON } = renderWithGlobalContext(<FeaturedMissions {...props} />, deps, initialState);
    expect(deps.apiService.fetchStreakList).toBeCalledTimes(1);
    await act(() => wait(0));
    expect(toJSON()).toMatchSnapshot();
  });
});
