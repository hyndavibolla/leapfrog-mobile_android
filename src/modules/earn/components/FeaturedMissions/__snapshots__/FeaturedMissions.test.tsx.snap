// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedMissions should refresh data by default 1`] = `
<View
  accessibilityLabel="earn-main-featured-missions-container"
  accessible={true}
  style={
    Object {
      "paddingBottom": 24,
      "paddingHorizontal": 16,
    }
  }
  testID="earn-main-featured-missions-container"
>
  <View
    accessibilityLabel="earn-main-featured-missions-streak-header"
    accessible={true}
    style={
      Object {
        "justifyContent": "space-between",
        "paddingBottom": 16,
      }
    }
    testID="earn-main-featured-missions-streak-header"
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#262C2F",
            "fontFamily": "SFProDisplay-Bold",
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
        ]
      }
      type="sectionHeader"
    >
      Featured Missions
    </Text>
  </View>
  <View>
    <View
      accessibilityLabel="large-streak-card-container"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="large-streak-card-container"
    >
      <View
        style={
          Object {
            "paddingBottom": 4,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "elevation": 3,
                "shadowColor": "#262C2F",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.1,
              },
              Object {
                "borderBottomLeftRadius": 8,
                "borderBottomRightRadius": 8,
                "borderTopLeftRadius": 8,
                "borderTopRightRadius": 8,
                "flexDirection": "column",
                "justifyContent": "space-between",
                "paddingBottom": 20,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 20,
              },
              Object {
                "borderBottomLeftRadius": 8,
                "borderBottomRightRadius": 8,
                "borderTopLeftRadius": 8,
                "borderTopRightRadius": 8,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderBottomLeftRadius": 8,
                  "borderBottomRightRadius": 8,
                  "borderTopLeftRadius": 8,
                  "borderTopRightRadius": 8,
                  "paddingBottom": 20,
                  "paddingLeft": 20,
                  "paddingRight": 20,
                  "paddingTop": 20,
                  "width": "100%",
                },
                Object {
                  "backgroundColor": "#EBEDF2",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View>
                <CMIFColorMatrixImageFilter
                  matrix={
                    Array [
                      0.2126,
                      0.7152,
                      0.0722,
                      0,
                      0,
                      0.2126,
                      0.7152,
                      0.0722,
                      0,
                      0,
                      0.2126,
                      0.7152,
                      0.0722,
                      0,
                      0,
                      0,
                      0,
                      0,
                      1,
                      0,
                    ]
                  }
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#fff0",
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "height": 60,
                          "width": 60,
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "overflow": "hidden",
                          },
                          Object {
                            "alignItems": "center",
                            "borderBottomLeftRadius": 60,
                            "borderBottomRightRadius": 60,
                            "borderTopLeftRadius": 60,
                            "borderTopRightRadius": 60,
                            "height": "100%",
                            "justifyContent": "center",
                            "width": "100%",
                          },
                        ]
                      }
                    >
                      <FastImageView
                        accessibilityLabel="brand-logo-image"
                        accessible={true}
                        onFastImageError={[Function]}
                        onFastImageLoad={[Function]}
                        onFastImageLoadEnd={[Function]}
                        onFastImageLoadStart={[Function]}
                        resizeMode="cover"
                        source={
                          Object {
                            "priority": "high",
                            "uri": "https://i.sears.com/s/i/or/SYW-MAX/Brands_Sam's-club-Logos-SYW-MAX_H5YHQFPMCT",
                          }
                        }
                        style={
                          Object {
                            "bottom": 0,
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                          }
                        }
                        testID="brand-logo-image"
                      />
                    </View>
                  </View>
                </CMIFColorMatrixImageFilter>
              </View>
              <View
                style={
                  Object {
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "marginLeft": 24,
                  }
                }
              >
                <Text
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "color": "#909BA0",
                        "fontFamily": "SFProDisplay-Medium",
                        "fontSize": 16,
                        "lineHeight": 22,
                      },
                      Object {
                        "color": "#262C2F",
                        "fontFamily": "SFProDisplay-Bold",
                        "fontSize": 16,
                      },
                    ]
                  }
                >
                  Grocery & Delivery Streak
                </Text>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "marginTop": 16,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "marginRight": 4,
                        }
                      }
                    >
                      <Text
                        accessibilityLabel="icon-text"
                        accessible={true}
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#0066CC",
                              "fontSize": 22,
                            },
                            undefined,
                            Object {
                              "fontFamily": "sywmax",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                        testID="icon-text"
                      >
                        î¥†
                      </Text>
                    </View>
                    <View
                      style={
                        Object {
                          "marginRight": 4,
                        }
                      }
                    >
                      <Text
                        accessibilityLabel="icon-text"
                        accessible={true}
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#0066CC",
                              "fontSize": 22,
                            },
                            undefined,
                            Object {
                              "fontFamily": "sywmax",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                        testID="icon-text"
                      >
                        î¥†
                      </Text>
                    </View>
                    <View
                      style={
                        Object {
                          "marginRight": 4,
                        }
                      }
                    >
                      <Text
                        accessibilityLabel="icon-text"
                        accessible={true}
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#0066CC",
                              "fontSize": 22,
                            },
                            undefined,
                            Object {
                              "fontFamily": "sywmax",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                        testID="icon-text"
                      >
                        î¥†
                      </Text>
                    </View>
                  </View>
                  <Text
                    accessibilityLabel="large-streak-card-complete"
                    accessible={true}
                    style={
                      Array [
                        Object {
                          "color": "#909BA0",
                          "fontFamily": "SFProDisplay-Medium",
                          "fontSize": 16,
                          "lineHeight": 22,
                        },
                        Object {
                          "color": "#0066CC",
                          "fontFamily": "SFProDisplay-Medium",
                          "fontSize": 14,
                        },
                      ]
                    }
                    testID="large-streak-card-complete"
                  >
                    Completed!
                  </Text>
                </View>
              </View>
            </View>
            <View
              accessibilityLabel="large-streak-card-ended"
              accessible={true}
              style={
                Object {
                  "marginBottom": 0,
                  "marginLeft": 20,
                  "marginRight": 20,
                  "marginTop": 24,
                }
              }
              testID="large-streak-card-ended"
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#909BA0",
                      "fontFamily": "SFProDisplay-Medium",
                      "fontSize": 16,
                      "lineHeight": 22,
                    },
                    Object {
                      "color": "#909BA0",
                      "fontFamily": "SFProDisplay-Medium",
                      "fontSize": 14,
                    },
                  ]
                }
              >
                Mission has ended. New mission coming soon!
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`FeaturedMissions should render 1`] = `
<View
  accessibilityLabel="earn-main-featured-missions-container"
  accessible={true}
  style={
    Object {
      "paddingBottom": 24,
      "paddingHorizontal": 16,
    }
  }
  testID="earn-main-featured-missions-container"
>
  <View
    accessibilityLabel="earn-main-featured-missions-streak-header"
    accessible={true}
    style={
      Object {
        "justifyContent": "space-between",
        "paddingBottom": 16,
      }
    }
    testID="earn-main-featured-missions-streak-header"
  >
    <Text
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#262C2F",
            "fontFamily": "SFProDisplay-Bold",
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
        ]
      }
      type="sectionHeader"
    >
      Featured Missions
    </Text>
  </View>
  <View>
    <View
      accessibilityLabel="large-streak-card-container"
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="large-streak-card-container"
    >
      <View
        style={
          Object {
            "paddingBottom": 4,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "elevation": 3,
                "shadowColor": "#262C2F",
                "shadowOffset": Object {
                  "height": 4,
                  "width": 0,
                },
                "shadowOpacity": 0.1,
              },
              Object {
                "borderBottomLeftRadius": 8,
                "borderBottomRightRadius": 8,
                "borderTopLeftRadius": 8,
                "borderTopRightRadius": 8,
                "flexDirection": "column",
                "justifyContent": "space-between",
                "paddingBottom": 20,
                "paddingLeft": 20,
                "paddingRight": 20,
                "paddingTop": 20,
              },
              Object {
                "borderBottomLeftRadius": 8,
                "borderBottomRightRadius": 8,
                "borderTopLeftRadius": 8,
                "borderTopRightRadius": 8,
                "paddingBottom": 0,
                "paddingLeft": 0,
                "paddingRight": 0,
                "paddingTop": 0,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#FFFFFF",
                  "borderBottomLeftRadius": 8,
                  "borderBottomRightRadius": 8,
                  "borderTopLeftRadius": 8,
                  "borderTopRightRadius": 8,
                  "paddingBottom": 20,
                  "paddingLeft": 20,
                  "paddingRight": 20,
                  "paddingTop": 20,
                  "width": "100%",
                },
                Object {
                  "backgroundColor": "#EBEDF2",
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View>
                <CMIFColorMatrixImageFilter
                  matrix={
                    Array [
                      0.2126,
                      0.7152,
                      0.0722,
                      0,
                      0,
                      0.2126,
                      0.7152,
                      0.0722,
                      0,
                      0,
                      0.2126,
                      0.7152,
                      0.0722,
                      0,
                      0,
                      0,
                      0,
                      0,
                      1,
                      0,
                    ]
                  }
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#fff0",
                      },
                      undefined,
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "height": 60,
                          "width": 60,
                        },
                        undefined,
                      ]
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "overflow": "hidden",
                          },
                          Object {
                            "alignItems": "center",
                            "borderBottomLeftRadius": 60,
                            "borderBottomRightRadius": 60,
                            "borderTopLeftRadius": 60,
                            "borderTopRightRadius": 60,
                            "height": "100%",
                            "justifyContent": "center",
                            "width": "100%",
                          },
                        ]
                      }
                    >
                      <FastImageView
                        accessibilityLabel="brand-logo-image"
                        accessible={true}
                        onFastImageError={[Function]}
                        onFastImageLoad={[Function]}
                        onFastImageLoadEnd={[Function]}
                        onFastImageLoadStart={[Function]}
                        resizeMode="cover"
                        source={
                          Object {
                            "priority": "high",
                            "uri": "https://i.sears.com/s/i/or/SYW-MAX/Brands_Sam's-club-Logos-SYW-MAX_H5YHQFPMCT",
                          }
                        }
                        style={
                          Object {
                            "bottom": 0,
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                          }
                        }
                        testID="brand-logo-image"
                      />
                    </View>
                  </View>
                </CMIFColorMatrixImageFilter>
              </View>
              <View
                style={
                  Object {
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "marginLeft": 24,
                  }
                }
              >
                <Text
                  ellipsizeMode="tail"
                  numberOfLines={1}
                  style={
                    Array [
                      Object {
                        "color": "#909BA0",
                        "fontFamily": "SFProDisplay-Medium",
                        "fontSize": 16,
                        "lineHeight": 22,
                      },
                      Object {
                        "color": "#262C2F",
                        "fontFamily": "SFProDisplay-Bold",
                        "fontSize": 16,
                      },
                    ]
                  }
                >
                  Grocery & Delivery Streak
                </Text>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "marginTop": 16,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flexDirection": "row",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "marginRight": 4,
                        }
                      }
                    >
                      <Text
                        accessibilityLabel="icon-text"
                        accessible={true}
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#0066CC",
                              "fontSize": 22,
                            },
                            undefined,
                            Object {
                              "fontFamily": "sywmax",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                        testID="icon-text"
                      >
                        î¥†
                      </Text>
                    </View>
                    <View
                      style={
                        Object {
                          "marginRight": 4,
                        }
                      }
                    >
                      <Text
                        accessibilityLabel="icon-text"
                        accessible={true}
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#0066CC",
                              "fontSize": 22,
                            },
                            undefined,
                            Object {
                              "fontFamily": "sywmax",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                        testID="icon-text"
                      >
                        î¥†
                      </Text>
                    </View>
                    <View
                      style={
                        Object {
                          "marginRight": 4,
                        }
                      }
                    >
                      <Text
                        accessibilityLabel="icon-text"
                        accessible={true}
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#0066CC",
                              "fontSize": 22,
                            },
                            undefined,
                            Object {
                              "fontFamily": "sywmax",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {},
                          ]
                        }
                        testID="icon-text"
                      >
                        î¥†
                      </Text>
                    </View>
                  </View>
                  <Text
                    accessibilityLabel="large-streak-card-complete"
                    accessible={true}
                    style={
                      Array [
                        Object {
                          "color": "#909BA0",
                          "fontFamily": "SFProDisplay-Medium",
                          "fontSize": 16,
                          "lineHeight": 22,
                        },
                        Object {
                          "color": "#0066CC",
                          "fontFamily": "SFProDisplay-Medium",
                          "fontSize": 14,
                        },
                      ]
                    }
                    testID="large-streak-card-complete"
                  >
                    Completed!
                  </Text>
                </View>
              </View>
            </View>
            <View
              accessibilityLabel="large-streak-card-ended"
              accessible={true}
              style={
                Object {
                  "marginBottom": 0,
                  "marginLeft": 20,
                  "marginRight": 20,
                  "marginTop": 24,
                }
              }
              testID="large-streak-card-ended"
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#909BA0",
                      "fontFamily": "SFProDisplay-Medium",
                      "fontSize": 16,
                      "lineHeight": 22,
                    },
                    Object {
                      "color": "#909BA0",
                      "fontFamily": "SFProDisplay-Medium",
                      "fontSize": 14,
                    },
                  ]
                }
              >
                Mission has ended. New mission coming soon!
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
